dashboard/app/models/user.rb

```diff
@@ -2075,6 +2077,32 @@ def destroy
     end
   end
 
+  # Called before_destroy.
+  # Soft-deletes any projects and other channel-backed progress belonging to
+  # this user.  Unfeatures any featured projects belonging to this user.
+  private def soft_delete_channels
+    return unless user_storage_id
+
+    channel_ids = PEGASUS_DB[:storage_apps].
+      where(storage_id: user_storage_id).
+      map(:id)
+
+    # Unfeature any featured projects owned by the user
+    FeaturedProject.
+      where(storage_app_id: channel_ids).
+      update_all(unfeatured_at: Time.now)
```

Comment:
might be worth chaining `.where.not(featured_at: nil)` before the `update_all` command, but that may not be important unless @Erin007 thinks it is.

Reply:
https://github.com/code-dot-org/code-dot-org/blob/ab84bcda7902f7a36e844753aebc77e3f40e3d50/dashboard/app/controllers/featured_projects_controller.rb#L15
As is, in the context of the table, we don't set featured_at to nil when unfeaturing because it's useful for Josh to be able to know when the project was most recently previously featured; so, I don't think it's necessary.

Reply:
Ah... but I might reasonably want to avoid updating `unfeatured_at` if a project is already unfeatured.  I'll adjust.